android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothCommunication.java

```diff
@@ -76,6 +79,19 @@ public BluetoothCommunication(Context context)
         this.central = new BluetoothCentral(context, bluetoothCentralCallback, new Handler(Looper.getMainLooper()));
     }
 
+    protected boolean needReConnect() {
+        if (callbackBtHandler == null) {
+            return true;
+        }
+        if (btPeripheral != null) {
```

Comment:
this can be simplified like:
`if(null == btPerfipheral)
          return true;

int state = btPeripheral.getState();
//....
return false;
}`

Reply:
@bcdiaconu Thanks for the comment!
If I understood you correctly then you would rewrite needReConnect() function like this:
```Java
    protected boolean needReConnect() {
        if (callbackBtHandler == null) {
            return true;
        }
        if (btPeripheral == null) {
            return true;
        }
        int state = btPeripheral.getState();
        if (state == BluetoothProfile.STATE_CONNECTED || state == BluetoothProfile.STATE_CONNECTING) {
            return false;
        }
        return true;
    }
```
On one hand I like that a level of nesting got removed, but on another hand I also liked that access to btPeripherial was nested inside null check `if`.

Reply:
yes. This is how I would do it. It is simpler to understand at human level. I would be checking for null references (and other exceptions that could arose) and treat them, than continue with normal logic. Generally speaking, this approach will lead to cleaner code and also to less execution than in case of checking and treating errors later on.

another thing I got used to be doing is to put the rvalue in the left of the equality operator. This will lead to less annoying to find bugs. that's why I wrote `if (null == lvalue)`. This way, if the equality operator is accidentally written as a assignment one (`if (null = lval)`) it will lead to compilation error. So this bug is found before it will misbehave. At the beginning seems weird but with just a bit of practice it comes naturally later on.

Reply:
About eliminating extra level of nesting - I agree with you. I won't change it right away (since this function is tiny), but definitely take it into account if it will grow. 

About putting `null` on the left side ... let's agree do disagree? My personal opinion is that although such approach might help prevent a miss-typing bug, I don't think it's worth sacrificing readability for that.  And Java won't let you do this with most data types/objects (because `if` requires a boolean).

Thanks again for your review! Please let me know If you find anything else!

Reply:
I see your point with the if null. At a more general level is about having the rvalue in the left side of the operator. Take this example: `if ( myInt = 5)` vs `if (5 = myInt)`. The first is compilable but second is not. And first will not lead to runtime exception either. it will simply return true.

with both the paragraphs, its your code and go with what you like. hopefully you will be taking it just as a helping observation.

Reply:
I was interested how you write code, because I encountered no errors in your patch and generally that is not the case for a first or second try. 

My two cents on a matter: I did not do a full review nor took a look in depth at logical level but I must say that for what I saw, you write beautiful code and easy to grasp.

Reply:
@bcdiaconu Thanks for the compliments!